api/v2beta1/helmrelease_types.go

```diff
@@ -653,10 +653,14 @@ type HelmReleaseStatus struct {
 	// +optional
 	LastAttemptedValuesChecksum string `json:"lastAttemptedValuesChecksum,omitempty"`
 
-	// LastReleaseRevision is the revision of the last successful Helm release.
+	// LastReleaseRevision is the revision of the last performed Helm release.
 	// +optional
 	LastReleaseRevision int `json:"lastReleaseRevision,omitempty"`
 
+	// LastSuccessfulReleaseRevision is the revision of the last successful Helm release.
+	// +optional
+	LastSuccessfulReleaseRevision int `json:"lastSuccessfulReleaseRevision,omitempty"`
```

Comment:
In this PR, this only seems to be used to tell if the `LastReleaseReivision` was successful, which seems like it could also be derived from the `Released` and `TestSuccess` conditions (and whether tests are enabled). Are there other reasons to add this field, such as for manual rollbacks?

Reply:
It is used to roll back to a revision we have made ourselves, see the change in the `Rollback` method.